### PR TITLE
backport all changes from `main` into `track/2.31`

### PR DESCRIPTION
**Pull request by DnPlas**
dex-auth 2.31 is still a valid version for the upcoming kubeflow 1.7 release, thus all the changes we have introduced in main should be backported into the track branch to keep it up to date.

Please note these changes have been already reviewed as they belong to the `main` branch and the way to know if they work is the CI passing.

### CODE REVIEW

File: metadata.yaml [hunk 1]
```diff
@@ -7,12 +7,68 @@ description: |
 requires:
```
Comment (i-chvets) [inline metadata.yaml:7]: @DnPlas Do we want to add these at this point?
```
website: https://charmhub.io/dex-auth
source: https://github.com/canonical/dex-auth-operator
issues: https://github.com/canonical/dex-auth-operator/issues
docs: https://discourse.charmhub.io/t/charmed-dex/8214
```

Comment (i-chvets) [reply metadata.yaml:7]: Otherwise LGTM

Comment (DnPlas) [reply metadata.yaml:7]: @i-chvets let's add it in a follow-up PR so both main and track/2.31 have the change.

Comment (i-chvets) [reply metadata.yaml:7]: +1

**Review by i-chvets — APPROVED** [review]
Approved.

**Pull request merged**